Enigmatry.Entry.SmartEnums/SmartEnumTypeExtensions.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Reflection;
+using Ardalis.SmartEnum;
+
+namespace Enigmatry.Entry.SmartEnums;
+
+// Taken from SmartEnum.EFCore
+public static class SmartEnumTypeExtensions
+{
+    public static IEnumerable<(Type EnumType, Type ValueType)> FindSmartEnums(
```

Comment:
Utility functions stolen from SmartEnum.EFCore, were needed for both JsonConverters registration and for SwaggerSchemaGeneration